src/commands/run.rs

```diff
@@ -0,0 +1,346 @@
+//! Implementation of the `run` subcommand.
+
+use std::path::Path;
+use std::path::PathBuf;
+use std::sync::Mutex;
+use std::time::Duration;
+
+use anyhow::Context;
+use anyhow::Result;
+use anyhow::bail;
+use clap::Parser;
+use colored::Colorize as _;
+use futures::FutureExt as _;
+use indexmap::IndexSet;
+use indicatif::ProgressStyle;
+use tokio::select;
+use tokio_util::sync::CancellationToken;
+use tracing::Level;
+use tracing::error;
+use tracing_indicatif::span_ext::IndicatifSpanExt as _;
+use wdl::ast::AstNode as _;
+use wdl::cli::Analysis;
+use wdl::cli::Evaluator;
+use wdl::cli::Inputs;
+use wdl::cli::analysis::Source;
+use wdl::cli::inputs::OriginPaths;
+use wdl::engine::EvaluationError;
+use wdl::engine::Inputs as EngineInputs;
+use wdl::engine::v1::ProgressKind;
+
+use crate::Mode;
+use crate::emit_diagnostics;
+
+/// The delay in showing the progress bar.
+///
+/// This is to prevent the progress bar from flashing on the screen for
+/// very short analyses.
+const PROGRESS_BAR_DELAY_BEFORE_RENDER: Duration = Duration::from_secs(2);
+
+/// Arguments to the `run` subcommand.
+#[derive(Parser, Debug)]
+#[clap(disable_version_flag = true)]
+pub struct Args {
+    /// A source WDL file or URL.
+    #[clap(value_name = "PATH or URL")]
+    pub source: Source,
+
+    /// The inputs passed in on the command line.
+    pub inputs: Vec<String>,
+
+    /// The name of the task or workflow to run.
+    ///
+    /// If inputs are provided, this will be attempted to be inferred from the
+    /// prefixed names of the inputs (e.g, `<name>.<input-name>`).
+    ///
+    /// If no inputs are provided and this argument is not provided, it will be
+    /// assumed you're trying to run the workflow present in the specified WDL
+    /// document.
+    #[clap(short, long, value_name = "NAME")]
+    pub name: Option<String>,
+
+    /// The task execution output directory; defaults to the task name.
```

Comment:
Is this the default behavior we want? Would something like `$PWD/sprocket/run/[timestamp]_$name/` be preferred?

Reply:
For now, I've kept the default of what Peter had in `wdl`. I think that's fine for the very first version, and we can obviously piecewise improve the process as we go along.